### PR TITLE
Add module pointers to env if they don't exist.

### PR DESCRIPTION
**Pull request by rohinb2**

### CODE REVIEW

**Comment by rohinb2** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/run-house/runhouse/914?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#914** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/914?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#909** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/909?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#908** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/908?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rohinb2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/functions/function_factory.py [hunk 1]
```diff
@@ -70,7 +70,9 @@ def function(
 
     fn_pointers = None
     if callable(fn):
-        fn_pointers = Function._extract_pointers(fn, reqs=env.reqs)
+        fn_pointers, req_to_add = Function._extract_pointers(fn, reqs=env.reqs)
+        if req_to_add:
+            env.reqs = [req_to_add] + env.reqs
```
Comment (dongreenberg) [inline runhouse/resources/functions/function_factory.py:75]: Do we need to check if it's already present?

Comment (dongreenberg) [reply runhouse/resources/functions/function_factory.py:75]: Or do we return None if it's already present?

**Review by dongreenberg — APPROVED** [review]

**Comment by rohinb2** [comment]
### Merge activity

* **Jun 25, 3:28 PM EDT**: @rohinb2 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/914).

**Pull request merged**